simd/src/Kokkos_SIMD_SVE.hpp

```diff
@@ -0,0 +1,3175 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_SIMD_SVE_HPP
+#define KOKKOS_SIMD_SVE_HPP
+
+#include <functional>
+#include <type_traits>
+
+#include <Kokkos_SIMD_Common.hpp>
+#include <Kokkos_SIMD_NEON.hpp>
+
+#ifdef KOKKOS_SIMD_COMMON_MATH_HPP
+#error \
+    "Kokkos_SIMD_SVE.hpp must be included before Kokkos_SIMD_Common_Math.hpp!"
+#endif
+
+#ifndef __ARM_FEATURE_SVE_BITS
+#error "Kokkos_SIMD_SVE.hpp: need to be compiled with -msve-vector-bits=<N>"
+#else
+
+#include <arm_sve.h>
+
+#ifdef __ARM_NEON_SVE_BRIDGE
+
+#if __has_include(<arm_neon_sve_bridge.h>)
+#include <arm_neon_sve_bridge.h>
+#else
+
+// NOTE: We observed that LLVM-14 and LLVM-15 defined __ARM_NEON_SVE_BRIDGE,
+// but not provided arm_neon_sve_bridge.h (!?)
+// https://godbolt.org/z/7jqWvzvWY
+KOKKOS_IMPL_WARNING(
+    "[WARNING]: You are using a compiler with __ARM_NEON_SVE_BRIDGE set, but \
+the header arm_neon_sve_bridge.h is missing. This could be a compiler \
+toolchain issue (observed on LLVM-14 and LLVM-15, see \
+https://godbolt.org/z/7jqWvzvWY). Kokkos will use its own implementation of \
+these functions, which could be slower than native ones in a recent SVE \
+compiler.")
```

Comment:
This looks really annoying. How would a user disable the warning after they acknowledged it's contents?

Reply:
The warning is managed by the existing macro `KOKKOS_IMPL_WARNING`. Unless Kokkos provide a yet-exist `KOKKOS_DISABLE_WARNING`, we can not hide them at compile time.